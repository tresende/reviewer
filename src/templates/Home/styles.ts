import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.main`
  display: flex;
  flex-direction: column;
`

export const Container = styled.div`
  ${media.lessThan('medium')`
    padding-right: 15px;
    padding-left: 15px;
  `}

  @media (min-width: 1200px) {
    width: 1170px;
  }

  @media (min-width: 992px) {
    width: 970px;
  }

  margin: 0 auto;
`

export const Main = styled.main`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    padding: ${theme.spacings.small} 0;
  `}
`

export const Menu = styled.section`
  ${({ theme }) => css`
    width: 100%;
    background-color: ${theme.colors.black};
  `}
`

export const Content = styled.section`
  ${({ theme }) => css`
    width: 100%;
    flex: 1;
    background-color: ${theme.colors.gray200};
  `}
`
