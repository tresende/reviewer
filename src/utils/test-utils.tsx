import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'
import { LoadingContext, LoadingContextData, LoadingContextDefaultValues } from 'hooks/use-loading'

type CustomRenderProps = {
  loadingProviderProps?: LoadingContextData
} & Omit<RenderOptions, 'queries'>

const customRender = (
  ui: ReactElement,
  { loadingProviderProps = LoadingContextDefaultValues, ...renderOptions }: CustomRenderProps = {}
) =>
  render(
    <ThemeProvider theme={theme}>
      <LoadingContext.Provider value={loadingProviderProps}>{ui}</LoadingContext.Provider>
    </ThemeProvider>,
    renderOptions
  )

export * from '@testing-library/react'
export { customRender as render }
