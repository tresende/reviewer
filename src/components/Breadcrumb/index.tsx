import * as S from './styles'

const Breadcrumb = () => (
  <S.Wrapper>
    <S.Item>
      <a href="#">BeerOrCoffee</a>
    </S.Item>
    <S.Item>
      <a href="#">Coworkings Brasil</a>
    </S.Item>
    <S.Item>
      <a href="#">BrasíS.Itema/DF</a>
    </S.Item>
    <S.Item>Asa Sul</S.Item>
    <S.Item>
      <a href="#">BSB Coworking</a>
    </S.Item>
    <S.Item current>Avaliações</S.Item>
  </S.Wrapper>
)

export default Breadcrumb
