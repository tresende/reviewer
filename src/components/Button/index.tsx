import { ButtonHTMLAttributes } from 'react'

import * as S from './styles'

export type ButtonProps = {
  children: React.ReactNode
} & ButtonHTMLAttributes<HTMLButtonElement>

const Button = ({ children, ...props }: ButtonProps) => <S.Wrapper {...props}>{children}</S.Wrapper>

export default Button
