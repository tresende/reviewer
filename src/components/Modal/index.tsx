import { useEffect } from 'react'

import * as S from './styles'

export type ModalProps = {
  open?: boolean
  children?: React.ReactNode
  onClose: () => void
}

const Modal = ({ onClose, children, open }: ModalProps) => {
  useEffect(() => {
    const handleKeyUp = ({ key }: KeyboardEvent) => {
      key === 'Escape' && onClose()
    }

    window.addEventListener('keyup', handleKeyUp)
    return () => window.removeEventListener('keyup', handleKeyUp)
  }, [])
  if (open === undefined) return null

  const modalClasses = `animate__animated ${!open ? 'animate__bounceOutUp' : 'animate__bounceInUp'} `
  const backgroundClasses = `animate__animated ${!open ? 'animate__fadeOut' : 'animate__fadeIn'} `

  return (
    <S.Wrapper open={open} onClick={onClose} className={backgroundClasses} role="modal">
      <S.Modal onClick={(e) => e.stopPropagation()} className={modalClasses}>
        {children}
      </S.Modal>
      {open && <S.GlobalStyle />}
    </S.Wrapper>
  )
}

export default Modal
