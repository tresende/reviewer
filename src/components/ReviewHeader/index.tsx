import * as S from './styles'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faStar } from '@fortawesome/free-solid-svg-icons'

const ReviewHeader = () => {
  library.add(faStar)
  return (
    <S.Wrapper>
      <S.Title>
        Avaliações de
        <a href="/coworking/space/bsb--brasilia">BSB Coworking</a>
        <span>Brasília/DF</span>
      </S.Title>
      <S.Divider />
      <div className="row">
        <div className="stars">
          <span className="icons-star">
            <FontAwesomeIcon size="1x" icon="star" />
          </span>
          <span className="num-review">5 avaliações</span>
        </div>
        <input
          id="writeReviewBtn"
          className="btn btn-block btn-outline btn-review pull-right"
          value="Avaliar"
          type="button"
        />
      </div>
    </S.Wrapper>
  )
}

export default ReviewHeader
