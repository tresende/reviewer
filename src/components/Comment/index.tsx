import Stars from 'components/Stars'
import Comment from 'models/Comment'
import { toPTBRFormat } from 'utils/date'

import * as S from './styles'

export type ButtonProps = {
  comment: Comment
}

const CommentComponent = ({ comment }: ButtonProps) => (
  <S.Wrapper>
    <S.Avatar src={comment.photo} />
    <S.Comment>
      <S.CommentHeader>
        <div>
          <S.Name>{comment.user}</S.Name>
          <S.SmallText>Avaliação verificada • </S.SmallText>
          <S.SmallText>{toPTBRFormat(comment.date)}</S.SmallText>
        </div>
        <Stars size={14} quantity={comment.score} />
      </S.CommentHeader>
      <S.Text>{comment.text}</S.Text>
    </S.Comment>
  </S.Wrapper>
)

export default CommentComponent
