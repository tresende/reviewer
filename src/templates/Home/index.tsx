import ReviewHeader from 'components/ReviewHeader'
import ReviewComments from 'components/ReviewComments'

import * as S from './styles'

const Home = () => (
  <S.Wrapper>
    <S.Menu />
    <S.Container>
      <S.Main>
        <ReviewHeader />
      </S.Main>
    </S.Container>
    <S.Content>
      <S.Container>
        <ReviewComments />
      </S.Container>
    </S.Content>
  </S.Wrapper>
)

export default Home
