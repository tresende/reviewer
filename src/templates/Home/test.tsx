import { render, screen } from 'utils/test-utils'
import Home from '.'
import Mock from 'mocks/mock'
import { ReviewHeaderProps } from 'components/ReviewHeader'
import { ModalProps } from 'components/Modal'

jest.mock('components/ReviewHeader', () => ({
  __esModule: true,
  default: function Mock({ onReview }: ReviewHeaderProps) {
    return <button onClick={onReview} data-testid="ReviewHeaderCallback" />
  }
}))

jest.mock('components/Modal', () => ({
  __esModule: true,
  default: function Mock({ onClose, open }: ModalProps) {
    return (
      <>
        {open && <span data-testid="ModalIsOpen" />}
        <button onClick={onClose} data-testid="ModalCallback" />
      </>
    )
  }
}))

describe('<Home />', () => {
  it('should open modal', () => {
    const { container } = render(<Home data={Mock} />)
    screen.getByTestId('ReviewHeaderCallback').click()

    expect(screen.getByTestId('ModalIsOpen')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should close modal', () => {
    render(<Home data={Mock} />)
    screen.getByTestId('ReviewHeaderCallback').click()
    screen.getByTestId('ModalCallback').click()

    expect(screen.queryByTestId('ModalIsOpen')).not.toBeInTheDocument()
  })
})
