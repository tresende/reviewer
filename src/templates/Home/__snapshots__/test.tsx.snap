// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should open modal 1`] = `
.c1 {
  max-width: 1100px;
  margin: 0 auto;
}

.c7 {
  display: none;
  cursor: pointer;
  background-color: #FABA40;
  margin: 0.8rem;
  padding: 0.8rem;
  border-radius: 0.4rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFFFFF;
  font-weight: 700;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c5 {
  text-align: center;
  font-family: 'Montserrat';
  font-size: 1.4rem;
  color: #FFFFFF;
  cursor: pointer;
}

.c9 {
  border: 2px solid #FABA40;
  border-radius: 30px;
  background-color: transparent;
  color: #FABA40;
  font-weight: 700;
  padding: 1.6rem;
}

.c11 {
  border: 2px solid #FFFFFF;
  border-radius: 30px;
  background-color: transparent;
  color: #FFFFFF;
  font-weight: 700;
  padding: 1.6rem;
}

.c10 {
  cursor: pointer;
  margin: 0.8rem;
  font-size: 1.4rem;
  font-weight: 700;
}

.c0 {
  background-color: #222222;
  padding: 1rem;
}

.c0 input {
  display: none;
}

.c0 .c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:768px) {
  .c7 {
    display: block;
    color: white;
    font-style: normal;
  }
}

@media (max-width:768px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:768px) {
  .c3 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 100%;
    padding: 0 1rem;
  }
}

@media (max-width:768px) {
  .c5 {
    padding: 0.8rem 0;
    border-bottom: 1px solid #4b4b4b;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    width: 100%;
    padding: 1rem;
    display: none;
  }
}

@media (max-width:768px) {
  .c6 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 0;
    border: 0;
  }
}

@media (max-width:768px) {
  .c9 {
    width: 100%;
    margin-top: 1rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 0.8rem;
    display: none;
  }
}

@media (max-width:768px) {
  .c11 {
    width: 100%;
    margin-top: 1rem;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    padding: 0.8rem;
    display: none;
  }
}

@media (max-width:768px) {
  .c10 {
    display: none;
  }
}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (max-width:768px) {

}

@media (min-width:1200px) {

}

@media (min-width:992px) {

}

<header
  class="c0"
>
  <nav
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4 c5 c6"
        >
          <img
            alt="Beer or Coffee"
            src="/img/logo.png"
          />
          <label
            class="c7"
            data-testid="open-menu"
          >
            ☰
          </label>
        </div>
        <div
          class="c5 animate__animated  "
        >
          Espaços de Coworking
        </div>
        <div
          class="c5 animate__animated  "
        >
          Planos
        </div>
        <div
          class="c5 animate__animated  "
        >
          Descontos
        </div>
      </div>
      <div
        class="c3 c8 animate__animated  "
      >
        <button
          class="c9"
        >
          Adiquirir um Plano
        </button>
        <span
          class="c10"
        >
          Empresas
        </span>
        <button
          class="c11"
          color="white"
        >
          Oi, Thiago
        </button>
      </div>
    </div>
  </nav>
</header>
`;
